pkg/apis/monitoring/v1/servicemonitor_types.go

```diff
@@ -83,6 +83,18 @@ type ServiceMonitorSpec struct {
 
 	// Label selector to select the Kubernetes `Endpoints` objects to scrape metrics from.
 	Selector metav1.LabelSelector `json:"selector"`
+
+	// Mechanism used to select the endpoints to scrape.
+	// By default, the selection process relies on relabel configurations to filter the discovered targets.
+	// Alternatively, you can opt in for role selectors, which may offer better efficiency in large clusters.
+	// Using role selectors might reduce Prometheus resource requirements Which strategy is best for your use case needs to be carefully evaluated.
```

Comment:
(nit) reading again we say the same thing twice.
```suggestion
	// Which strategy is best for your use case needs to be carefully evaluated.
```